### PR TITLE
Install sqlalchemy-spanner package into Google provider

### PR DESCRIPTION
**Pull request by MaksYermak**
In this PR I have added sqlalchemy-spanner package to Google provider and updated the Google Spanner hook for supporting  DbApiHook.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/spanner.py [hunk 1]
```diff
@@ -70,6 +77,34 @@ def _get_client(self, project_id: str) -> Client:
             )
         return self._client
 
+    def _get_conn_params(self) -> tuple:
+        """Extract spanner database connection parameters."""
+        extras = self.get_connection(self.gcp_conn_id).extra_dejson
+        project_id = get_field(extras, "project_id") or self.project_id
+        instance_id = get_field(extras, "instance_id")
+        database_id = get_field(extras, "database_id")
```
Comment (moiseenkov) [inline airflow/providers/google/cloud/hooks/spanner.py:85]: Are these fields always there? Is their absence possible and what happens in this case?

File: airflow/providers/google/cloud/hooks/spanner.py [hunk 2]
```diff
@@ -70,6 +77,34 @@ def _get_client(self, project_id: str) -> Client:
             )
         return self._client
 
+    def _get_conn_params(self) -> tuple:
```
[the remark this answers is not in the document]
Comment (MaksYermak) [reply airflow/providers/google/cloud/hooks/spanner.py:80]: I think it's a good idea I will add it

File: airflow/providers/google/cloud/hooks/spanner.py [hunk 3]
```diff
@@ -70,6 +77,34 @@ def _get_client(self, project_id: str) -> Client:
...
+        return project_id, instance_id, database_id
+
+    def get_uri(self) -> str:
+        """Override DbApiHook get_uri method for get_sqlalchemy_engine()."""
+        project_id, instance_id, database_id = self._get_conn_params()
+        if not instance_id or not database_id:
+            raise AirflowException("The instance_id or database_id were not specify")
+        return f"spanner+spanner:///projects/{project_id}/instances/{instance_id}/databases/{database_id}"
+
+    def get_sqlalchemy_engine(self, engine_kwargs=None):
```
[the remark this answers is not in the document]
Comment (MaksYermak) [reply airflow/providers/google/cloud/hooks/spanner.py:95]: I am not sure that we should add type here. In all cases when this method is implemented I have seen them without type. In my opinion it's better to leave like this and then we will see what the community will say about it.

File: airflow/providers/google/cloud/hooks/spanner.py (hunk 1 above)
Comment (MaksYermak) [reply airflow/providers/google/cloud/hooks/spanner.py:85]: If their absence then these variables are None and code below throws Exception.

Comment (moiseenkov) [reply airflow/providers/google/cloud/hooks/spanner.py:85]: LGTM, thanks

File: tests/providers/google/cloud/hooks/test_spanner.py [hunk 4]
```diff
@@ -431,6 +433,12 @@ def test_execute_dml_overridden_project_id(self, get_client):
         run_in_transaction_method.assert_called_once_with(mock.ANY)
         assert res is None
 
+    def test_get_sqlalchemy_engine(self):
+        self.spanner_hook_default_project_id._get_conn_params = MagicMock(return_value=SPANNER_CONN_PARAMS)
+        engine = self.spanner_hook_default_project_id.get_sqlalchemy_engine()
+        assert isinstance(engine, sqlalchemy.engine.Engine)
+        assert engine.name == "spanner+spanner"
```
[the remark this answers is not in the document]
Comment (MaksYermak) [reply tests/providers/google/cloud/hooks/test_spanner.py:449]: I don't think it's necessary. We check that the `engine` is created, also `create_engine()` function can have a big amount parameters passed on through kwargs.

**Review by moiseenkov — APPROVED** [review]
LGTM, but there are some unit tests failing, and it might be related to these changes. Perhaps, it makes sense to check them:
```
TestGcpSpannerHookDefaultProjectId::test_get_sqlalchemy_engine
TestGcpSpannerHookNoDefaultProjectID::test_get_sqlalchemy_engine
```

**Pull request closed without merging**